### PR TITLE
doc: explain script

### PR DESCRIPTION
**Pull request by adhusson**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,11 +85,48 @@ To run specific test suites, use the `testSuites` package script:
...
+
+- Name its contract (multiple instances of the same contract may be deployed)
+- Ignore script names (different scripts are used for different networks)
+- Have all its deployed contracts in one place
+
+To do the above, Mangrove adds a layer to `forge script` deployment.
+
+- Deployment scripts should inherit the `Deployer` contract.
+- You should call `outputDeployment()` at the end of your scripts.
+  - When `outputDeployment()` gets called, a file all known deployed contracts are written to `addresses/deployed.backup/<network>-latest.json`
```
Comment (lnist) [inline README.md]: file with all?

File: README.md [hunk 2]
```diff
@@ -85,11 +85,48 @@ To run specific test suites, use the `testSuites` package script:
...
+- You should set `WRITE_DEPLOY=true` when running scripts that you want to broadcast.
```
Comment (lnist) [inline README.md:109]: and --broadcast?

File: README.md [hunk 3]
```diff
@@ -85,11 +85,48 @@ To run specific test suites, use the `testSuites` package script:
...
+(Note: in this context, etherscan can mean "polygonscan" or any block explorer)
+
+## Chain-dependent broadcast with private keys in .env
+
+The `vm.broadcast()` cheatcode implicitly selects a sender for the broadcast transactions: either the default sender, or the address associated to the `--private-key` given in the CLI, or to `--mnemonic` information, etc.
+
+It is tiring to always add `--private-key 0x..` to scripts, especially since the key may be different for each network. The `Deployer` contract has a `broadcast()` function that reads the `<NAME>_PRIVATE_KEY` var off the environment, where `NAME` is the name of the chain you're talking to.
+
+- You should have vars such as `MUMBAI_PRIVATE_KEY` , `POLYGON_PRIVATE_KEY` in your `.env` file
+- You should use `broadcast()` instead of `vm.broadcast()`. The deployer contract will look for the correct private key, and fallback to the CLI-provided key if none was found.
```
Comment (lnist) [inline README.md]: you should use... in scripts.

File: README.md [hunk 4]
```diff
@@ -85,11 +85,48 @@ To run specific test suites, use the `testSuites` package script:
...
+
+To do the above, Mangrove adds a layer to `forge script` deployment.
+
+- Deployment scripts should inherit the `Deployer` contract.
+- You should call `outputDeployment()` at the end of your scripts.
+  - When `outputDeployment()` gets called, a file all known deployed contracts are written to `addresses/deployed.backup/<network>-latest.json`
+- You should set `WRITE_DEPLOY=true` when running scripts that you want to broadcast.
+  - When `WRITE_DEPLOY=true`, the contract set is also written to `addresses/deployed/<network>.json`.
+
+(Note that for mumbai, `network=maticmum`)
```
Comment (lnist) [inline README.md:112]: why not mumbai? :)

**Review by lnist — COMMENTED** [review]
minor comments/questions

File: README.md (hunk 2 above)
Comment (adhusson) [reply README.md:109]: Yes but I'm describing the added layer not the normal forge script how-to

File: README.md (hunk 4 above)
Comment (adhusson) [reply README.md:112]: Because ethers calls it maticmum :( so it's consistent all the way to mangrove.js.

**Review by lnist — APPROVED** [review]

**Pull request merged**